### PR TITLE
Add magics recipe

### PR DESCRIPTION
**Pull request by kynan**
ECMWF's Meteorological plotting software

@kmuehlbauer @ocefpaf @carletes

### CODE REVIEW

**Comment by kynan** [comment]
Is this good to go @conda-forge/core ?

File: recipes/magics/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    - boost 1.63.*
+    - proj.4
+    - python-eccodes
+    - python 2.7.*
+    # ecBuild fails to find the C++ bindings: the library name is libnetcdf_c++4
+    # - netcdf-cxx4
+  run:
+    - proj.4
+    - python-eccodes
+    - python 2.7.*
```
Comment (ocefpaf) [inline recipes/magics/meta.yaml]: Update your skip statement to `skip: True  # [win or not py27]` and remove this pin here and above.

File: recipes/magics/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  number: 0
+  skip: True  # [win or py3k]
+  detect_binary_files_with_prefix: true
+
+requirements:
+  build:
+    - cmake
+    - swig
+    - boost 1.63.*
+    - proj.4
```
Comment (ocefpaf) [inline recipes/magics/meta.yaml]: Take a look at the pinnings. I believe this one must be pinned.

File: recipes/magics/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    # - netcdf-cxx4
+
+test:
+  commands:
+    - metgram
+
+about:
+  home: https://software.ecmwf.int/wiki/display/MAGP/
+  license: Apache 2.0
+  summary: ECMWF's Meteorological plotting software
```
Comment (ocefpaf) [inline recipes/magics/meta.yaml]: Is the license file in the tarball? If so let us add it here.

File: recipes/magics/meta.yaml (hunk 2 above)
Comment (kynan) [inline recipes/magics/meta.yaml]: How do I check this?

Comment (ocefpaf) [inline recipes/magics/meta.yaml]: Apologies. I assumed you had experience with conda-forge pinnings :flushed: 

Take a look at https://github.com/conda-forge/conda-forge.github.io/blob/master/scripts/pin_the_slow_way.py#L39-L82

(I am working on better docs for this one.)

**Comment by kynan** [comment]
Thanks for the review @ocefpaf! I believe all concerns are now addressed.

**Comment by kynan** [comment]
@ocefpaf [seems the package has not been built](https://github.com/conda-forge/magics-feedstock), or has it just not been built *yet*. How can I see the CI build status of a new feedstock?

File: recipes/magics/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  skip: True  # [win or not py27]
+  detect_binary_files_with_prefix: true
+
+requirements:
+  build:
+    - cmake
+    - swig
+    - boost 1.64.*
+    - proj.4 4.9.3
+    - python-eccodes
```
Comment (isuruf) [inline recipes/magics/meta.yaml]: `python-eccodes` is python 2.7 only and this library is python 3 only. So this library is not built by conda-build

Comment (kynan) [inline recipes/magics/meta.yaml]: No, it's the other way round: this library is also python 2.7 only.

Comment (isuruf) [inline recipes/magics/meta.yaml]: Okay, found the issue. https://github.com/conda-forge/proj.4-feedstock/pull/15
`proj.4` needs to be renamed to `proj4`

Comment (ocefpaf) [inline recipes/magics/meta.yaml]: Argh. Just always bites us b/c both `defaults` and `conda-forge` used to name it `proj.4`...

Comment (kynan) [inline recipes/magics/meta.yaml]: Thanks, I'll correct it on the feedstock!

**Pull request merged**